internal/controllers/language_detect/language_detect.go

```diff
@@ -235,3 +237,58 @@ func (ld *LanguageDetect) isBatFileOrShellFile(lang string) bool {
 	return strings.EqualFold(lang, "Batchfile") ||
 		strings.EqualFold(lang, "Shell")
 }
+
+// This method was added because if o project analyzed is submodule is not possible run tool gitleaks,
+// then was added this logic for check if .git in cloned folder in .horusec is a file.
+// Case yes, we can find where is original git folder and replace in .horusec folder for run gitleaks tool
+// without problems.
+// Case is a folder we can go to next step without breaking changes.
+func (ld *LanguageDetect) copyGitFolderWhenIsSubmodule(directory, folderDstName string) error {
+	if ld.configs.EnableGitHistoryAnalysis {
+		isGitSubmodule, originalFolderPath := ld.returnGitFolderOriginalIfIsSubmodule(filepath.Join(directory, ".git"))
+		if isGitSubmodule {
+			logger.LogErrorWithLevel(messages.MsgErrorCopyProjectToHorusecAnalysis,
+				os.RemoveAll(filepath.Join(folderDstName, ".git")))
+			err := copy.Copy(filepath.Join(directory, originalFolderPath), filepath.Join(folderDstName, ".git"),
+				func(src string) bool { return false })
+			if err != nil {
+				logger.LogErrorWithLevel(messages.MsgErrorCopyProjectToHorusecAnalysis, err)
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+//nolint:funlen // lines is not necessary broken
+func (ld *LanguageDetect) returnGitFolderOriginalIfIsSubmodule(directory string) (bool, string) {
+	fileInfo, err := os.Stat(directory)
+	if err != nil {
+		logger.LogError(messages.MsgErrorCopyProjectToHorusecAnalysis, err)
+		return false, ""
+	}
+	if fileInfo.IsDir() {
+		return false, ""
+	}
+	fileContentBytes, err := os.ReadFile(directory)
+	if err != nil {
+		logger.LogError(messages.MsgErrorCopyProjectToHorusecAnalysis, err)
+		return false, ""
+	}
+	return ld.validateSubModuleContent(fileContentBytes)
+}
+
+func (ld *LanguageDetect) validateSubModuleContent(fileContentBytes []byte) (bool, string) {
+	var fileContent string
+	const PrefixSubModule = "gitdir: "
+	if fileContentBytes != nil {
+		fileContent = string(fileContentBytes)
+	}
+	if len(fileContent) < len(PrefixSubModule) {
+		logger.LogErrorWithLevel(messages.MsgErrorCopyProjectToHorusecAnalysis,
+			errors.New("file content wrong: "+fileContent))
+		return false, ""
+	}
+
+	return true, fileContent[8 : len(fileContent)-1]
```

Comment:
What is actually returned here?

Reply:
Path where contains git folder in horusec examples o content of the file `./examples/.git` is:
```
gitdir: ../.git/modules/examples
```

Also i remove prefix `gitdir: ` and join with project path for copy .git folder correctly inside of .horusec

Reply:
What do you think put this information as a doc on code? Seems a trick that could be easily forgotten

Reply:
added